src/Components/Generator/DynamicInputContainer.js

```diff
@@ -96,7 +96,7 @@ export default class DynamicInputContainer extends preact.Component {
         return (
             <IntlProvider scope="generator" definition={I18N_DEFINITION}>
                 <div className="dynamic-input-container">
-                    <h2>{this.props.title}</h2>
+                    <h2 style={this.props.margin}>{this.props.title}</h2>
```

Comment:
I think this should rather be solved by adding a class on a flag. The name of the attribute here is misleading (`heading-syle` would be more appropriate) and `margin-top: 0` is already set via a more general style. Simply add a class `has-margin` to the heading and then add the style to the SCSS at `.box h2.has-margin`.